connectors/agent/config.py

```diff
@@ -0,0 +1,105 @@
+from connectors.config import add_defaults
+
+
+class ConnectorsAgentConfigurationWrapper:
+    """A wrapper that facilitates passing configuration from Agent to Connectors Service.
+
+    This class is responsible for:
+    - Storing in-memory configuration of Connectors Service running on Agent
+    - Transforming configuration reported by Agent to valid Connectors Service configuration
+    - Indicating that configuration has changed so that the user of the class can trigger the restart
+    """
+
+    def __init__(self):
+        """Inits the class.
+
+        There's default config that allows us to run connectors natively (see _force_allow_native flag),
+        when final configuration is reported these defaults will be merged with defaults from Connectors
+        Service config and specific config coming from Agent.
+        """
+        self._default_config = {
+            "_force_allow_native": True,
+            "native_service_types": [
+                "azure_blob_storage",
+                "box",
+                "confluence",
+                "dropbox",
+                "github",
+                "gmail",
+                "google_cloud_storage",
+                "google_drive",
+                "jira",
+                "mongodb",
+                "mssql",
+                "mysql",
+                "notion",
+                "onedrive",
+                "oracle",
+                "outlook",
+                "network_drive",
+                "postgresql",
+                "s3",
+                "salesforce",
+                "servicenow",
+                "sharepoint_online",
+                "slack",
+                "microsoft_teams",
+                "zoom",
+            ],
+        }
+
+        self.specific_config = {}
+
+    def try_update(self, source):
+        """Try update the configuration and see if it changed.
+
+        This method takes the check-in event coming from Agent and checks if config needs an update.
+
+        If update is needed, configuration is updated and method returns True. If no update is needed
+        the method returns False.
+        """
+
+        # TODO: find a good link to what this object is.
+        has_hosts = source.fields.get("hosts")
+        has_api_key = source.fields.get("api_key")
+        has_basic_auth = source.fields.get("username") and source.fields.get("password")
+        if has_hosts and (has_api_key or has_basic_auth):
+            es_creds = {
+                "host": source["hosts"][0],
+            }
+
+            if source.fields.get("api_key"):
+                es_creds["api_key"] = source["api_key"]
```

Comment:
@artem-shelkovnikov commenting so that it doesn't get lost, by default we would get Fleet-specific api key format, this should do the trick to turn it to our version:

```
    if source.fields.get("api_key"):
        api_key = source["api_key"]
        # if beats_logstash_format
        if ":" in api_key:
            api_key = base64.b64encode(api_key.encode()).decode()
```

Reply:
Verified that this works

Reply:
Thanks for the investigation! I'm gonna add this code as well :)